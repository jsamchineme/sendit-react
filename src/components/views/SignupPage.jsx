import React from 'react';
import { NavLink } from 'react-router-dom';
import SignupForm from '../forms/SignupForm';

const LoginPage = () => {
  return (
    <div className="wrapper">
      <section className="page-content">
        <section className="mast no-bg paddless">
          <NavLink to='/' className="logo-text-group">
            <div className="logo"><img src="assets/img/sendit-pickup-ondemand.svg" alt="sendit-logo" /></div>
            <div className="text">Send<span>IT</span></div>
          </NavLink>
        </section>
        <div className="login-box">
          <SignupForm />
        </div>
      </section>
      <div className="page-background">
        <div className="overlay" />
      </div>
    </div>
  );
}

export default LoginPage;