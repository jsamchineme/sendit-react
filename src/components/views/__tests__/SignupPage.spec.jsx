import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import { BrowserRouter, Route } from 'react-router-dom';
import { SignupPage } from '../SignupPage';
import configureStore from '../../../redux/configureStore';

const store = configureStore();

describe('Tests for signup page', () => {
  it('should render the signup page', () => {
    const props = {
      form: {
        loginForm: {
          registeredFields: {
            email: {
              name: 'email',
              type: 'Field',
              count: 1
            },
            password: {
              name: 'password',
              type: 'Field',
              count: 1
            }
          },
          fields: {
            email: {
              visited: true,
              touched: true
            },
            password: {
              visited: true,
              touched: true
            }
          },
          values: {
            email: 'gasg@g.com',
            password: 'secreatpass'
          },
          anyTouched: true
        }
      },
    }
    const subject = (
      <SignupPage {...props} />
    );

    const wrapper = shallow(subject);

    expect(wrapper).toMatchSnapshot();
  });
});