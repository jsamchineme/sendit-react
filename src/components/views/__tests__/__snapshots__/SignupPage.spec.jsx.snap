// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for signup page should render the signup page 1`] = `
<div
  className="wrapper"
>
  <section
    className="page-content"
  >
    <section
      className="mast no-bg paddless"
    >
      <NavLink
        className="logo-text-group"
        to="/"
      >
        <div
          className="logo"
        >
          <img
            alt="sendit-logo"
            src="/assets/img/sendit-pickup-ondemand.svg"
          />
        </div>
        <div
          className="text"
        >
          Send
          <span>
            IT
          </span>
        </div>
      </NavLink>
    </section>
    <div
      className="login-box"
    >
      <ReduxForm
        destroyOnUnmount={true}
        enableReinitialize={false}
        forceUnregisterOnUnmount={false}
        form="signupForm"
        getFormState={[Function]}
        handleSubmit={[Function]}
        keepDirtyOnReinitialize={false}
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
      />
    </div>
  </section>
  <div
    className="page-background"
  >
    <div
      className="overlay"
    />
  </div>
</div>
`;
